src/state/versions.js

```diff
@@ -0,0 +1,75 @@
+import * as api from 'services/api';
+
+export const initialState = {
+  loading: false,
+  error: null,
+  bblfshdVersion: undefined,
+  dashboardVersion: undefined,
+  loadedFrom: null,
+};
+
+export const LOAD = 'bblfsh/versions/LOAD';
+export const SET = 'bblfsh/versions/SET';
+export const FAILED = 'bblfsh/versions/FAILED';
+
+export const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case LOAD:
+      return {
+        ...state,
+        loading: true,
+        error: null,
+        loadedFrom: action.from,
+      };
+    case SET:
+      return {
+        ...state,
```

Comment:
the lines below this one overrides the whole state, doesn't it? If so,`...state` is no longer needed here.
An alternative may be:
```javascript:
return {
  ...initialState,
  bblfshdVersion: action.bblfshdVersion,
  dashboardVersion: action.dashboardVersion,
}
```
because it seems the `SET` action resets the state to the `initialState.loading` and `initialState.error`

Reply:
the good practice is to take state and modify it as needed, without throwing away. As soon as we add one more property in state and it won't be set by `set` action - we will have to refactor to current state.

Reply:
make sense